### DISCUSSION TITLE
Using isValid only applicable to onChange Event?

### DISCUSSION THREAD

**Discussion by tempoeric** [react-hook-form/react-hook-form#5994] [General]
Hi does using isValid only applicable to onChange? Does it not apply to schema that has no errors like pre populated checkbox?
![image](https://user-images.githubusercontent.com/42777871/126073368-1af35228-d7ba-4ae0-95eb-fb73739bc1d0.png)
Based on the screenshot I provided after clicking my radio button to Yes and the checkbox value has already been checked due to antd's default checked value, isValid does not validate the schema and giving me a false value which should be true cause i have no error.

This is the yup schema validation im using for my checkbox:
```
accountFacility: Yup.array()
      .when('withFacilities', {
        is: true,
        then: Yup.array()
          .required('ErrorMessage.Required')
          .min(1, 'ErrorMessage.Required'),
      })
      .nullable(),

```

**Comment by tempoeric** [comment]
Please notice this @bluebill1049. Thanks. Since this is a major blocker for me right now

**Reply by bluebill1049** [reply]
I would investigate your resolver, hook form basically send ur form values to your schema.

**Reply by tempoeric** [reply]
Hi @bluebill1049 is there a way to validate the form in onLoad? 
```

const {
    clearErrors,
    control,
    handleSubmit,
    reset,
    setValue,
    watch,
    formState: { errors,  isValid, isSubmitSuccessful },
  } = useForm({
    defaultValues: accountDetails,
    resolver: yupResolver(
      schema(
        crossBorderRemittanceRefCode,
        isSelectedSavingsProduct,
        jointHolderList,
        otherRefCode,
        relatedRefCode,
        isChecking,
      ),
    ),
    mode: 'onChange',
  });
```

**Reply by bluebill1049** [reply]
`trigger`?

**Reply by tempoeric** [reply]
can i ask how to use trigger im only new to react hooks

**Reply by bluebill1049** [reply]
```
React.useEffect(() => {
  trigger()
}, [trigger])
```

look up react doc for useEfffect and trigger at doc for hook form.

**Reply by tempoeric** [reply]
trigger didnt help still need the section to be dirty or onChange to be validated by isValid

**Reply by tempoeric** [reply]
hi @bluebill1049 any updates on this one?
